### DISCUSSION TITLE
How to do retries on failed mutations?

### DISCUSSION THREAD

**Discussion by dmahajan980** [TanStack/query#4594] [Q&A] [answered]
Hello, I'm creating a mutation which gets fired when the user saves their data. However, given the nature of our backend service, it might fail sometimes and I would like the mutation to retry when it fails. The behaviour that I'm expecting sounds the same when queries fail (auto-retries with exponentially increasing time gaps). Is this possible to achieve this in mutations (`useMutation`) without having to write your own code?

Apologies for my ignorance if this info has been mentioned already somewhere and thanks in advance for taking the time to help me out :)

**Comment by TkDodo** [comment, marked as answer]
mutations also support the `retry` option - it just defaults to zero, but you can set it however you want. It's the same option as for queries (`retry` and `retryDelay`).

**Reply by dmahajan980** [reply]
Thank you very much :)
